lighthouse-core/gather/driver.js

```diff
@@ -816,10 +868,8 @@ class Driver {
     }).then(_ => {
       return async () => {
         log.warn('Driver', 'Timed out waiting for page load. Checking if page is hung...');
-        waitForFCP.cancel();
-        waitForLoadEvent.cancel();
-        waitForNetworkIdle.cancel();
-        waitForCPUIdle.cancel();
+        waitForSecurityCheck.cancel();
```

Comment:
I think it'd be easier to reason about all this canceling if each cancelable promise stores state about if it's been canceled yet and avoids doing it more than once. Then we could judiciously use `cancelAll` in one place, after the Promise race.

Reply:
Yeah this is ripe for cleanup, but this new parallel style will make it so much easier now, thank you!!

Reply:
FWIW, I don't think anything would be harmed today if you threw this into `cancelAll` right?

Reply:
would also have to remove the self-cancel that `_waitForSecurityCheck` does, otherwise it would cancel itself twice.

I'll try the cancel state thing first

Reply:
hang on before you do that I was saying I don't think it would matter if we called `cancel` twice, we're just extra disabling the security domain and removing a listener that might've been removed already.

Reply:
OK, LMK if that is better or worse